### PR TITLE
[vcpkg scripts] Fix check the different features have the same feature variable in function vcpkg_check_features

### PR DESCRIPTION
**Pull request by JonLiu1993**
Fixes `vcpkg_check_features`scripts:
````
    list(SORT feature_variables)
    set(last_variable)
    foreach(variable IN LISTS feature_variables)
        if(variable STREQUAL last_variable)
            message(FATAL_ERROR "vcpkg_check_features passed the same feature variable multiple times: '${variable}'")
        endif()
    endforeach()
````
Here need to add `set(last_variable ${variable})`

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [ ] ~~SHA512s are updated for each updated download~~
- [ ] ~~The "supports" clause reflects platforms that may be fixed by this new version~~
- [ ] ~~Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.~~
- [ ] ~~Any patches that are no longer applied are deleted from the port's directory.~~
- [ ] ~~The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.~~
- [ ] ~~Only one version is added to each modified port's versions file.~~

### CODE REVIEW

**Comment by dg0yt** [comment]
Please adjust the title (for proper change records): It is not `checkout`.

**Comment by JonLiu1993** [comment]
@vicroms, I have tested successfully locally. If we don’t add `set(last_variable ${variable})`, when two different features have duplicate `${variable}`, the error cannot be triggered. After adding set(last_variable `${variable})` The error can be triggered successfully.
Portfile.cmake:
````
.....
vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS
    FEATURES
        hwopt   WITH_OPTIMIZATIONS
        tool    WITH_OPTIMIZATIONS
)

vcpkg_cmake_configure(
    SOURCE_PATH "${SOURCE_PATH}"
    OPTIONS
        ${FEATURE_OPTIONS}
        -DUPSTREAM_VER=${VERSION}
    OPTIONS_DEBUG
        -DBUILD_TOOL=OFF
)
.....
````
````
PS F:\test\vcpkg> ./vcpkg install argon2[*]:x64-windows
Computing installation plan...
The following packages will be built and installed:
    argon2[core,hwopt,tool]:x64-windows -> 20190702#1
Detecting compiler hash for triplet x64-windows...
Restored 0 package(s) from C:\Users\v-zhli17\AppData\Local\vcpkg\archives in 299 us. Use --debug to see more details.
Installing 1/1 argon2:x64-windows...
Building argon2[core,hwopt,tool]:x64-windows...
-- Using cached P-H-C-phc-winner-argon2-20190702.tar.gz.
-- Cleaning sources at F:/test/vcpkg/buildtrees/argon2/src/20190702-1481e5839f.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source F:/test/vcpkg/downloads/P-H-C-phc-winner-argon2-20190702.tar.gz
-- Applying patch visibility.patch
-- Applying patch visibility-for-tool.patch
-- Applying patch thread-header.patch
-- Using source at F:/test/vcpkg/buildtrees/argon2/src/20190702-1481e5839f.clean
CMake Error at scripts/cmake/vcpkg_check_features.cmake:90 (message):
  vcpkg_check_features passed the same feature variable multiple times:
  'WITH_OPTIMIZATIONS'
Call Stack (most recent call first):
  ports/argon2/portfile.cmake:15 (vcpkg_check_features)
  scripts/ports.cmake:147 (include)


error: building argon2:x64-windows failed with: BUILD_FAILED
Please ensure you're using the latest port files with `git pull` and `vcpkg update`.
Then check for known issues at:
    https://github.com/microsoft/vcpkg/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+argon2
You can submit a new issue at:
    https://github.com/microsoft/vcpkg/issues/new?title=[argon2]+Build+error&body=Copy+issue+body+from+F%3A%2Ftest%2Fvcpkg%2Finstalled%2Fvcpkg%2Fissue_body.md
````
The pipline error has nothing to do with my PR modification, I think this PR can be merged directly.

**Comment by dg0yt** [comment]
> The pipline error has nothing to do with my PR modification, I think this PR can be merged directly.

CI showed that at least one port has the bug detected by this check. You must fix it as part of the change, or we get a baseline regression.

**Comment by JonLiu1993** [comment]
I fix the port bug for `mnn ogre libsbml`.

**Review by dg0yt — CHANGES_REQUESTED** [review]
LGTM, but for ogre I disagree.

File: ports/ogre/vcpkg.json [hunk 1]
```diff
@@ -87,12 +103,21 @@
...
+        },
         "python3"
       ]
     },
     "strict": {
       "description": "Force Ogre resource manager to be strict with group names and resource names"
     },
+    "swig": {
+      "description": "find package swig"
+    },
```
[the remark this answers is not in the document]
Comment (JonLiu1993) [reply ports/ogre/vcpkg.json]: Done.

File: scripts/cmake/vcpkg_check_features.cmake [hunk 2]
```diff
@@ -89,6 +89,7 @@ function(vcpkg_check_features)
         if(variable STREQUAL last_variable)
             message(FATAL_ERROR "vcpkg_check_features passed the same feature variable multiple times: '${variable}'")
```
Comment (BillyONeal) [inline scripts/cmake/vcpkg_check_features.cmake:90]:
```suggestion
            message("${Z_VCPKG_BACKCOMPAT_MESSAGE_LEVEL}" "vcpkg_check_features passed the same feature variable multiple times: '${variable}'")
```

Given that it was toasted before I think we need to let existing callers of old versions continue to work.

**Review by LilyWangLL — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks for shutting down this bug!

**Pull request merged**